Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM gcc:7
+MAINTAINER Abdul Dakkak <dakkak@illinois.edu>
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        build-essential \
+        automake \
+        autoconf \
+        bison \
+        flex \
+        libevent-dev \
+        libboost-all-dev \
+        libssl-dev \
+        libcurl4-openssl-dev \
+        libdouble-conversion-dev \
+        libgoogle-glog-dev \
+        libgflags-dev \
+        libjemalloc-dev \
+        libssl-dev \
+        cmake \
+        thrift-compiler  && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+
+ENV FOLLY_SRC=/src/folly
+RUN git clone https://github.com/facebook/folly $FOLLY_SRC && cd $FOLLY_SRC && git fetch --all
```

Comment:
Could you use a environment variable (USE_BUNDLED_FOLLY or USE_GIT_FOLLY?) to define to use a versioned package or git branch?

You could refer to cmake [external project files](https://github.com/flier/zipkin-cpp/tree/master/cmake), it will try to download and build [folly](https://github.com/flier/zipkin-cpp/blob/master/cmake/InstallFolly.cmake) if necessary.